examples/insert-node.rs

```diff
@@ -0,0 +1,52 @@
+// Shows how to maintain nodes sorted by value in a
+// machine-generated kdl document
+
+use kdl::{KdlDocument, KdlIdentifier, KdlNode};
+
+fn sort_by_name(x: &KdlNode, y: &KdlNode) -> std::cmp::Ordering {
+    x.name().value().cmp(y.name().value())
+}
+
+fn main() -> miette::Result<()> {
+    let input = r#"
+words {
+  apple
+  orange
+}
+    "#;
+    let mut doc: KdlDocument = input.parse()?;
+
+    let words_section = doc.get_mut("words").expect("'words' section should exist");
+    let children = words_section
+        .children_mut()
+        .as_mut()
+        .expect("'words' section should have children");
+    let word_nodes = children.nodes_mut();
+
+    let identifier = KdlIdentifier::from("banana");
+    let word_node = KdlNode::new(identifier);
+    word_nodes.push(word_node);
+
+    word_nodes.sort_by(sort_by_name);
+
+    // Make sure all word nodes have the same indentation (2 spaces)
+    for (i, word) in word_nodes.iter_mut().enumerate() {
+        if i == 0 {
+            word.set_leading("\n  ");
+        } else {
+            word.set_leading("  ");
+        }
+        word.set_trailing("\n");
+    }
```

Comment:
Because I was not sure if it would do want i wanted.

Also, about 1/3 of the time when my IDE auto-completed fmt, it got the version with one argument from std instead.

You may want to rename the `fmt` method, by the way.

Also also, I expected `fmt()` to be recursive, which is not the case IIRC

Reply:
I'm on helix + rust-analyzer and unfortunately I did not manage to reproduce the issue. (If I do, I'll open a separate issue).

You're right, using `fmt()` works as long as you're OK with 4-space indentation :) Updated the example.